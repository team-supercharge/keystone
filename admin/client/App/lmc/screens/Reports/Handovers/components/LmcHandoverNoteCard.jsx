import React from 'react'
import PropTypes from 'prop-types'

const LmcHandoverNoteCard = ({ note }) => {
    const carer = note.carer

    return (
        <div 
            style={styles.noteContainer}
        >
            <div style={styles.carerContainer}>
                <div 
                    className="lmc-profile-picture__handover__small" 
                    style={{
                        background: `url(${carer.picture || PLACEHOLDER_IMAGE})` 
                    }} 
                />
                <span style={styles.carerName}>
                    { `${carer.name.first} ${carer.name.last}` }
                </span>
            </div>
        </div>
    )
}

const PLACEHOLDER_IMAGE = 'https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png';

const styles = {
    carerContainer: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'flex-start'
    },
    carerName: {
        fontSize: 14,
        opacity: 0.6,
        position: 'relative',
        top: 6,
        left: 15,
        margin: 0
    },
    noteContainer: {
        background: 'white',
        borderRadius: 10,
        border: '1px #999999 solid',
        width: '100%',
    },
}

LmcHandoverNoteCard.propTypes = {
    note: PropTypes.object.isRequired
}

export default LmcHandoverNoteCard