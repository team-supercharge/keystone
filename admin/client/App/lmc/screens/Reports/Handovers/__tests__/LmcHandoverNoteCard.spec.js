import React from 'react'
import { shallow } from 'enzyme'
import LmcHandoverNoteCard from '../components/LmcHandoverNoteCard.jsx'

describe('LmcHandoverNoteCard', () => {
    let wrapper
    let note

    beforeEach(() => {
        note = { 
            id: 'id', 
            carer: { name: { first: 'Test', last: 'Carer' }, picture: 'TestPicture' }, 
            createdOn: new Date('1/1/2019') 
        }

        wrapper = shallow(
            <LmcHandoverNoteCard
                note={note}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
