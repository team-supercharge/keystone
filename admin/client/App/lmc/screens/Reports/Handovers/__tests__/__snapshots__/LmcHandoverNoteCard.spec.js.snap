// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcHandoverNoteCard renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "background": "white",
        "border": "1px #999999 solid",
        "borderRadius": 10,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <div
        className="lmc-profile-picture__handover__small"
        style={
          Object {
            "background": "url(TestPicture)",
          }
        }
      />
      <span
        style={
          Object {
            "fontSize": 14,
            "left": 15,
            "margin": 0,
            "opacity": 0.6,
            "position": "relative",
            "top": 6,
          }
        }
      >
        Test Carer
      </span>
    </div>
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "background": "white",
          "border": "1px #999999 solid",
          "borderRadius": 10,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <div
          className="lmc-profile-picture__handover__small"
          style={
            Object {
              "background": "url(TestPicture)",
            }
          }
        />
        <span
          style={
            Object {
              "fontSize": 14,
              "left": 15,
              "margin": 0,
              "opacity": 0.6,
              "position": "relative",
              "top": 6,
            }
          }
        >
          Test Carer
        </span>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LmcHandoverNoteCard
        note={
          Object {
            "carer": Object {
              "name": Object {
                "first": "Test",
                "last": "Carer",
              },
              "picture": "TestPicture",
            },
            "createdOn": 2019-01-01T00:00:00.000Z,
            "id": "id",
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "note": Object {
            "carer": Object {
              "name": Object {
                "first": "Test",
                "last": "Carer",
              },
              "picture": "TestPicture",
            },
            "createdOn": 2019-01-01T00:00:00.000Z,
            "id": "id",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "background": "white",
              "border": "1px #999999 solid",
              "borderRadius": 10,
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <div
              className="lmc-profile-picture__handover__small"
              style={
                Object {
                  "background": "url(TestPicture)",
                }
              }
            />
            <span
              style={
                Object {
                  "fontSize": 14,
                  "left": 15,
                  "margin": 0,
                  "opacity": 0.6,
                  "position": "relative",
                  "top": 6,
                }
              }
            >
              Test Carer
            </span>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "background": "white",
              "border": "1px #999999 solid",
              "borderRadius": 10,
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <div
              className="lmc-profile-picture__handover__small"
              style={
                Object {
                  "background": "url(TestPicture)",
                }
              }
            />
            <span
              style={
                Object {
                  "fontSize": 14,
                  "left": 15,
                  "margin": 0,
                  "opacity": 0.6,
                  "position": "relative",
                  "top": 6,
                }
              }
            >
              Test Carer
            </span>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcHandoverNoteCard
    note={
      Object {
        "carer": Object {
          "name": Object {
            "first": "Test",
            "last": "Carer",
          },
          "picture": "TestPicture",
        },
        "createdOn": 2019-01-01T00:00:00.000Z,
        "id": "id",
      }
    }
  />,
}
`;
